### PR TITLE
Fix service account downgrades workspace admin to viewer

### PR DESCRIPTION
**Pull request by kushalmalani**
## Description

> Describe the purpose of this pull request.
Fixing mutation to create rolebinding without assigned UserId to a new deployment service account, since this is causing the workspace admin (non system admin) user to downgrade and have view access under deployment level.

## 🎟 Issue(s)

Resolves astronomer/issues#1901

## 🧪 Functional Testing

> List the functional testing steps to confirm this feature or fix.

## 📸 Screenshots

> Add screenshots to illustrate the validity of these changes.

## 📋 Checklist

- [x] Rebased from the main (or release if patching) branch (before testing)
- [x] Ran `yarn lint && yarn test` before taking out of draft
- [x] Added/updated applicable tests
- [ ] Tested against [Astro-UI](https://github.com/astronomer/astro-ui/) and [Astro-CLI](https://github.com/astronomer/astro-cli/) clients.
- [ ] Communicated to/tagged owners of respective clients potentially impacted by these changes.
- [ ] Updated any related [documentation](https://github.com/astronomer/docs/)

### CODE REVIEW

File: src/resolvers/mutation/create-deployment-service-account/index.js [hunk 1]
```diff
@@ -25,8 +25,7 @@ export default async function createDeploymentServiceAccount(
       roleBinding: {
         create: {
           role,
-          deployment: { connect: { id: deploymentUuid } },
-          user: { connect: { id: ctx.user.id } }
+          deployment: { connect: { id: deploymentUuid } }
```
Comment (vishwas-astro) [inline src/resolvers/mutation/create-deployment-service-account/index.js:28]: @kushalmalani trying to understand the syntax here. Is this basically creating a record without userId but only serviceAccountID and deploymentUUID?

Also, any chance we can add a test.

Comment (kushalmalani) [reply src/resolvers/mutation/create-deployment-service-account/index.js:28]: @vishwas-astro Yes, creating a rolebinding with userId is overrding the user deployment role. We already have a test covering this scenario. We didnt have one with userId.

is there a specific scenario you want a test on?

Comment (vishwas-astro) [reply src/resolvers/mutation/create-deployment-service-account/index.js:28]: Basically, after the operation,  ensure that there are 2 role bindings, one for user ID/depl Id and another for service acct Id/depl Id.

**Review by andriisoldatenko — CHANGES_REQUESTED** [review]
Please add practical and unit tests to make sure there is no regression.

File: src/resolvers/mutation/create-deployment-service-account/index.js (hunk 1 above)
Comment (andriisoldatenko) [reply src/resolvers/mutation/create-deployment-service-account/index.js:28]: @kushalmalani try to add something like:
```
expectx(expect).toHaveBeenCalledWith({without user})
```
also try to create new deployment service account/old service account and check if you can do astro deploy/or basic astro cli operations.

**Comment by andriisoldatenko** [comment]
@kushalmalani i still don't understand the change... i created SA before and after your change:
```
postgres=# select "deploymentId", "userId" from "RoleBinding" where "deploymentId" = 'ckg6ioeng03270jxtbrht0glw';
       deploymentId        |          userId
---------------------------+---------------------------
 ckg6ioeng03270jxtbrht0glw |
 ckg6ioeng03270jxtbrht0glw | ckg6io5b500070jxtzhhki9u9
(2 rows)
```

**Comment by andriisoldatenko** [comment]
@kushalmalani can you please demonstrate which query returns less data before and after you change?

**Comment by andriisoldatenko** [comment]
@kushalmalani make sense!

it's just typo in 0.20:
`it diff release-0.20 release-0.18 -- src/resolvers/mutation/create-deployment-service-account/index.js`

```diff
diff --git src/resolvers/mutation/create-deployment-service-account/index.js src/resolvers/mutation/create-deployment-service-account/index.js
index 60c75da4..bab84020 100644
--- src/resolvers/mutation/create-deployment-service-account/index.js
+++ src/resolvers/mutation/create-deployment-service-account/index.js
@@ -1,3 +1,6 @@
+import fragment from "./fragment";
+import validateExistence from "service-accounts/existence";
+import { addFragmentToInfo } from "graphql-binding";
 import crypto from "crypto";
 
 /*
@@ -10,11 +13,15 @@ import crypto from "crypto";
 export default async function createDeploymentServiceAccount(
   parent,
   args,
-  ctx
+  ctx,
+  info
 ) {
   // Pull out some variables.
   const { label, category, deploymentUuid, role } = args;
 
+  // Validate if service account label exists or not
+  await validateExistence({ label, deploymentUuid });
+
   // Create the base mutation.
   const mutation = {
     data: {
@@ -25,13 +32,17 @@ export default async function createDeploymentServiceAccount(
       roleBinding: {
         create: {
           role,
-          deployment: { connect: { id: deploymentUuid } },
-          user: { connect: { id: ctx.user.id } }
+          deployment: {
+            connect: { id: deploymentUuid }
+          }
         }
       }
     }
   };
 
   // Run the mutation.
-  return ctx.prisma.serviceAccount.create(mutation);
+  return ctx.db.mutation.createServiceAccount(
+    mutation,
+    addFragmentToInfo(info, fragment)
+  );
 }

```

**Review by andriisoldatenko — APPROVED** [review]

File: src/resolvers/mutation/create-deployment-service-account/index.unit.test.js [hunk 2]
```diff
@@ -69,8 +69,9 @@ describe("createDeploymentServiceAccount", () => {
 
     // Run the graphql mutation.
     const res = await graphql(schema, mutation, null, { prisma, user }, vars);
-
+    const expected = res.data.createDeploymentServiceAccount;
     expect(res.errors).toBeUndefined();
     expect(create.mock.calls).toHaveLength(1);
+    expect(expected.user).toBeFalsy();
```
Comment (adam2k) [inline src/resolvers/mutation/create-deployment-service-account/index.unit.test.js:75]: @kushalmalani @andriisoldatenko let me know if this is the correct test for this mutation.  This should never return a user object, correct?

File: src/resolvers/mutation/create-deployment-service-account/index.unit.test.js [hunk 3]
```diff
@@ -69,8 +69,9 @@ describe("createDeploymentServiceAccount", () => {
 
     // Run the graphql mutation.
     const res = await graphql(schema, mutation, null, { prisma, user }, vars);
-
+    const actual = res.data.createDeploymentServiceAccount;
     expect(res.errors).toBeUndefined();
     expect(create.mock.calls).toHaveLength(1);
+    expect(actual.user).toBeFalsy();
```
Comment (adam2k) [inline src/resolvers/mutation/create-deployment-service-account/index.unit.test.js:75]: @kushalmalani @andriisoldatenko let me know if this is correct.  There shouldn't be a user object for any service account, correct?

Comment (kushalmalani) [reply src/resolvers/mutation/create-deployment-service-account/index.unit.test.js:75]: Yeah, this looks ok

Comment (adam2k) [reply src/resolvers/mutation/create-deployment-service-account/index.unit.test.js:75]: If you want to we could also validate that all the values come back correctly that are mocked out in the create const, but that might not be necessary.

**Review by adam2k — APPROVED** [review]
Looks good to me! 🎉

File: src/resolvers/mutation/create-deployment-service-account/index.unit.test.js (hunk 3 above)
Comment (kushalmalani) [reply src/resolvers/mutation/create-deployment-service-account/index.unit.test.js:75]: its a mock right...its going to return whatever we send, regardless.

**Pull request merged**